src/test/java/de/tum/in/www1/artemis/metis/PostIntegrationTest.java

```diff
@@ -73,18 +86,24 @@ public void initTestCase() {
 
         student1 = database.getUserByLogin("student1");
 
-        // initialize test setup and get all existing posts (there are 4 posts with lecture context, 4 with exercise context, and 3 with course-wide context - initialized): 11
+        // initialize test setup and get all existing posts (there are 4 posts with lecture context, 4 with exercise context, 1 plagiarism case and 3 with course-wide context -
+        // initialized): 11
         // posts in total
         existingPosts = database.createPostsWithinCourse();
 
+        existingCoursePosts = existingPosts.stream().filter(post -> post.getPlagiarismCase() == null).collect(Collectors.toList());
```

Comment:
Use .toList() here and below, see #5312
```suggestion
        existingCoursePosts = existingPosts.stream().filter(post -> post.getPlagiarismCase() == null).toList();
```